docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,296 @@
+
+If your Application Service is using the EdgeX MessageBus trigger, you can simply copy the complete `Trigger` configuration from the example below and tweak it as needed. 
+
+!!! example "Example - EdgeX MessageBus trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "redis"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        SubscribeTopics="edgex/events/#"
+        [Trigger.EdgexMessageBus.PublishHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        PublishTopic="example"
+        [Trigger.EdgexMessageBus.Optional]
+        AuthMode = 'usernamepassword'  # required for redis messagebus (secure or insecure).
+        SecretName = 'redisdb'
+    ```
+
+From the above example you can see the improved structure and the following changes:
+
+- Default `EdgexMessageBus` type has changed to `Redis` from `ZeroMQ`
+- Type value for `Redis` has changed from `redistreams` to `redis`. This is because the implementation no longer uses Redis Streams. It now using Redis Pub/Sub.
+- `SubscribeTopics` is now plural since it now accepts a comma separated list of topics. The default value uses a multi-level topic with a wild card. This is because Core Data and Device Services now publish to a multi-level topics which have`edgex/events` as their base. This allows Application Services to filter by topic rather then receive the data and then filter it out via a pipeline filter function. See the [Filter By Topics](../Triggers/#filter-by-topics) section for more details.
+- The EdgeX MessageBus using Redis is a Secure MessageBus, thus the addition of the `AuthMode` and `SecretName` settings which allow the credentials to be pulled from the service's SecretStore. See the [Secure MessageBus](../../../security/Ch-Secure-MessageBus) secure for more details.
+
+##### External MQTT
+
+If your Application service is using the **External MQTT** trigger do the following:
+
+1. Move your existing `MqttBroker` configuration under the `Trigger` section (renaming it to `ExternalMqtt`)
+2. Move your `SubscribeTopic` (renaming it to `SubscribeTopics`) under the `ExternalMqtt` section.
+3. Move your `PublishTopic` under the `ExternalMqtt` section.
+
+!!! example "Example - External MQTT trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="external-mqtt"
+      [Trigger.ExternalMqtt]
+      Url = "tcp://broker.hivemq.com:1883"
+      SubscribeTopics = "edgex-trigger"
+      PublishTopic = "edgex-trigger-response"
+      ClientId = "app-my-service"
+      ConnectTimeout = "30s"
+      AutoReconnect = false
+      KeepAlive = 60
+      QoS = 0
+      Retain = false
+      SkipCertVerify = false
+      SecretPath = ""
+      AuthMode = "none"
+    ```
+
+##### HTTP
+
+The HTTP trigger configuration has not changed beyond the renaming of `Binding` to `Trigger`
```

Comment:
add a period here as well? There may be other sentences you need to add periods to later in the doc to remain consistent.